specification/ai/OpenAI.Assistants/assistants/routes.tsp

```diff
@@ -0,0 +1,159 @@
+  listAssistants(...OpenAIListRequestOptions): OpenAIPageableListOf<Assistant>;
+
+  /**
+   * Retrieves an existing assistant.
+   *
+   * @param assistantId The ID of the assistant to retrieve.
+   * @returns The requested assistant instance.
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "not yet an Azure operation"
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version" "not yet versioned"
+  #suppress "@azure-tools/typespec-azure-core/use-standard-names" "mirrored API name parity"
+  @get
+  @route("{assistantId}")
+  @added(ServiceApiVersions.v2023_11_06_beta)
+  retrieveAssistant(@path assistantId: string): Assistant;
+
+  /**
+   * Modifies an existing assistant.
+   *
+   * @param assistantId The ID of the assistant to modify.
+   * @param modificationOptions The details of the modification to perform on the specified assistant.
+   * @returns The updated assistant instance.
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "not yet an Azure operation"
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version" "not yet versioned"
+  @post
+  @route("{assistantId}")
+  @added(ServiceApiVersions.v2023_11_06_beta)
+  modifyAssistant(
+    @path assistantId: string,
+    @body modificationOptions: AssistantModificationOptions,
+  ): Assistant;
+
+  /**
+   * Deletes an assistant.
+   *
+   * @param assistantId The ID of the assistant to delete.
+   * @returns Status information about the requested deletion operation.
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "not yet an Azure operation"
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version" "not yet versioned"
+  @delete
+  @route("{assistantId}")
+  @added(ServiceApiVersions.v2023_11_06_beta)
+  deleteAssistant(@path assistantId: string): AssistantDeletionStatus;
+
+  /**
+   * Attaches a previously uploaded file to an assistant for use by tools that can read files.
+   *
+   * @param assistantId The ID of the assistant to attach the file to.
+   * @param fileId The ID of the previously uploaded file to attach.
+   * @returns Information about the attached file.
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "not yet an Azure operation"
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-core/operation-missing-api-version" "not yet versioned"
+  @post
+  @route("{assistantId}/files")
+  @added(ServiceApiVersions.v2023_11_06_beta)
+  createAssistantFile(
+    @path assistantId: string,
+    @projectedName("json", "file_id") fileId: string,
```

Comment:
Should this be annotated with `@body` ? Not sure what the generator behaviour is in this particular case, when a basic type is passed instead of a model file.

Reply:
The use of `@body` would effectively splat the type "as" the body, implying the content_type should match the type of the decorated body field, in this case making it "text/plain". We still want this to be a JSON request with a named `file_id` key and this is the way to represent that. For comparison, see a bit earlier with `modifyAssistant`, where `@body` *does* perform that desired splat (but, because `AssistantModificationOptions` is a model, we're still left with JSON).